rabbitmqclient/cluster_reference_test.go

```diff
@@ -534,6 +533,142 @@ var _ = Describe("ParseReference", func() {
 			})
 		})
 	})
+
+	Context("namespace permissions", func() {
+		BeforeEach(func() {
+			existingRabbitMQCluster = &rabbitmqv1beta1.RabbitmqCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rmq",
+					Namespace: namespace,
+				},
+				Status: rabbitmqv1beta1.RabbitmqClusterStatus{
+					Binding: &corev1.LocalObjectReference{
+						Name: "rmq-default-user-credentials",
+					},
+					DefaultUser: &rabbitmqv1beta1.RabbitmqClusterDefaultUser{
+						ServiceReference: &rabbitmqv1beta1.RabbitmqClusterServiceReference{
+							Name:      "rmq",
+							Namespace: namespace,
+						},
+					},
+				},
+			}
+			existingCredentialSecret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rmq-default-user-credentials",
+					Namespace: namespace,
+				},
+				Data: map[string][]byte{
+					"username": []byte(existingRabbitMQUsername),
+					"password": []byte(existingRabbitMQPassword),
+				},
+			}
+			existingService = &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rmq",
+					Namespace: namespace,
+				},
+				Spec: corev1.ServiceSpec{
+					ClusterIP: "1.2.3.4",
+					Ports: []corev1.ServicePort{
+						{
+							Name: "management",
+							Port: int32(15672),
+						},
+					},
+				},
+			}
+		})
+
+		When("requested namespace is prohibited", func() {
+			BeforeEach(func() {
+				existingRabbitMQCluster.ObjectMeta.Annotations = map[string]string{}
+				objs = []runtime.Object{existingRabbitMQCluster, existingCredentialSecret, existingService}
+			})
+			It("should return an error about a cluster being prohibited", func() {
+				_, _, err := rabbitmqclient.ParseReference(ctx, fakeClient,
+					topology.RabbitmqClusterReference{
+						Name:      existingRabbitMQCluster.Name,
+						Namespace: existingRabbitMQCluster.Namespace,
+					},
+					"prohibited-namespace",
+					"")
+				Expect(err).To(MatchError(rabbitmqclient.ResourceNotAllowedError))
+			})
+		})
+
+		When("there is a list of allowed namespaces", func() {
+			BeforeEach(func() {
+				existingRabbitMQCluster.ObjectMeta.Annotations = map[string]string{
+					"rabbitmq.com/topology-allowed-namespaces": "allowed1,allowed2",
+				}
+				objs = []runtime.Object{existingRabbitMQCluster, existingCredentialSecret, existingService}
+			})
+			When("requested namespace is in topology-allowed-namespaces annotation", func() {
```

Comment:
I'd prefer not having implementation details in the test description. As of now, this test reads as "context namespace permissions, when there is a list of allowed namespaces, and when requested namespace is in topology-allowed-namespaces annotation, it works". There is an implementation detail (annotation name) in the description. This implementation detail is not necessary to explain the behaviour.

IMO, it would read better as "context namespace permissions, when there is a list of allowed namespaces, and when requested namespace is allowed, it works". In this way, we would describe the scenario without tying its description to the annotation name.